library/modules/Gui.cpp

```diff
@@ -2151,16 +2152,31 @@ df::coord Gui::getMousePos()
     df::coord pos;
     if (gps && gps->precise_mouse_x > -1) {
         pos = getViewportPos();
-/* TODO: understand how this changes for v50
-        pos.x += gps->mouse_x_pixel / tile_width;
-        pos.y += gps->mouse_y_pixel / tile_height;
-*/
+        if (Screen::inGraphicsMode()) {
+            int32_t map_tile_pixels = gps->viewport_zoom_factor >> 2;
```

Comment:
Maybe `/ 4` would be more intuitive? Hopefully not slower, assuming these are all integers.

Reply:
fair enough. I guess I got into the habit of bit shifting when I could back when it mattered